### PR TITLE
draft: validation 

### PR DESCRIPTION
**Pull request by tom-wa**

### CODE REVIEW

File: doc/decisions/validation.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Validation
+
+## Issue
+
+Validation plugins operate as indenpendent blackboxes. 
+For every backend each mounted validation plugin iterates 
+over the whole keyset, checks every key for it's trigger metakey, 
+and validates the key.
+
+Each plugin decides on it's own what should happen when a key
```
Comment (markus2330) [inline doc/decisions/validation.md]: `its`

File: doc/decisions/validation.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+## Issue
+
+Validation plugins operate as indenpendent blackboxes. 
+For every backend each mounted validation plugin iterates 
+over the whole keyset, checks every key for it's trigger metakey, 
+and validates the key.
+
+Each plugin decides on it's own what should happen when a key 
+fails to validate.
```
Comment (markus2330) [inline doc/decisions/validation.md]: and the issue is? (missing flexibility&need to mount all validation plugins)

File: doc/decisions/validation.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+Each plugin decides on it's own what should happen when a key 
+fails to validate. 
+
+## Constraints
+
+## Assumptions
+
+While plugins should always fail and return
+an error if validation fails on kdbSet, there are be different
+requirements on what should happen on kdbGet:
```
Comment (markus2330) [inline doc/decisions/validation.md]: some rationale why we would need this. I do not think all of these usecases actually are useful.

File: doc/decisions/validation.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+of each key to the corresponding validation plugin. 
+
+## Argument
+
+- Validation plugins don't need to know what should be done if the validation fails
+- We can run multiple validations on every key and improve error messages
+- Different ways of handling errors only need to be implemented once
+
+## Implications
+
```
Comment (markus2330) [inline doc/decisions/validation.md]: We need to export `validate`.

File: doc/decisions/validation.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+- Extend validation plugins to allow us to specify what should happen 
+  if a key fails to validate
+- Export a validation function that allows us to use an additional plugin 
+  to decide what should be done
+
+
+## Decision
+
+Use a wapper plugin to iterate over the keyset and delegate the validation 
+of each key to the corresponding validation plugin.
```
Comment (markus2330) [inline doc/decisions/validation.md]: The default `kdbGet` behavior is not described.

**Comment by markus2330** [comment]
@tom-wa Please add `ready to merge` when review comments are fixed.

File: doc/decisions/validation.md (hunk 5 above)
Comment (tom-wa) [inline doc/decisions/validation.md]: i'm not really sure you want / expect here

**Pull request merged**